website/addons/mendeley/model.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+
+import time
+
+import mendeley
+from modularodm import fields
+
+from website import settings
+from website.addons.base import AddonNodeSettingsBase
+from website.addons.base import AddonUserSettingsBase
+from website.oauth.models import ExternalProvider
+
+from . import utils
+from .api import APISession
+
+
+def serialize_folder(name, account_id, parent_id=None, list_id=None, id=None):
+    retval = {
+        'name': name,
+        'provider_account_id': account_id,
+        'provider_list_id': list_id,
+        'id': id
+    }
+    if parent_id:
+        retval['parent_list_id'] = parent_id
+
+    return retval
+
+
+class AddonMendeleyUserSettings(AddonUserSettingsBase):
+    def _get_connected_accounts(self):
+        """Get user's connected Mendeley accounts"""
+        return [
+            x for x in self.owner.external_accounts if x.provider == 'mendeley'
+        ]
+
+    def to_json(self, user):
+        ret = super(AddonMendeleyUserSettings, self).to_json(user)
+        ret['accounts'] = [
+            utils.serialize_account(each)
+            for each in self._get_connected_accounts()
+        ]
+        return ret
+
+
+class AddonMendeleyNodeSettings(AddonNodeSettingsBase):
+    external_account = fields.ForeignField('externalaccount',
+                                           backref='connected')
+
+    mendeley_list_id = fields.StringField()
+
+    # Keep track of all user settings that have been associated with this
+    # instance. This is so OAuth grants can be checked, even if the grant is
+    #   not currently being used.
+    associated_user_settings = fields.AbstractForeignField(list=True)
+
+    _api = None
+
+    @property
```

Comment:
This makes a write to the database, so it might not be appropriate for it to be a property.